File: jscomp/test/floatarray_test.js
'use strict';

var Mt = require("./mt.js");
var Caml_array = require("../../lib/js/caml_array.js");

var suites = {
  contents: /* [] */0
};

var test_id = {
  contents: 0
};

function eq(loc, x, y) {
  return Mt.eq_suites(test_id, suites, loc, x, y);
}

var v = Caml_array.caml_make_float_vect(5);

for(var i = 0; i <= 4; ++i){
  v[i] = 0;
}

Caml_array.set(v, 2, 15.5);

eq("File \"floatarray_test.ml\", line 17, characters 5-12", [
      v.length,
      v[2],
      Caml_array.get(v, 1)
    ], [
      5,
      15.5,
      0
    ]);

Mt.from_pair_suites("Floatarray_test", suites.contents);

var K;

exports.suites = suites;
exports.test_id = test_id;
exports.eq = eq;
exports.K = K;
/* v Not a pure module */

File: jscomp/test/floatarray_test.ml
let suites : Mt.pair_suites ref = ref []
let test_id = ref 0
let eq loc x y = Mt.eq_suites ~test_id ~suites loc x y

module K = Array.Floatarray

let () =
  let len = 5 in
  let v = K.create len in
  for i = 0 to len - 1 do
    K.unsafe_set v i 0.
  done ;
  K.set v 2 0x1.fp3 ;
  eq __LOC__ (K.length v, K.unsafe_get v 2, K.get v 1) (len, 0x1.fp3, 0.)

let () = Mt.from_pair_suites __MODULE__ !suites
